pemFioi/conceptViewer-1.0-mobileFirst.js

```diff
@@ -470,6 +466,43 @@ function getConceptsFromBlocks(includeBlocks, allConcepts, context) {
   return concepts;
 }
 
+/**
+ * This function allow us to remove useless concept from the concept list.
+ * The "useless concepts" are concept who don't have a "python" list of all
+ * their functions or in which this list is empty.
+ * @param allConcepts All of the concepts.
+ * @return A new list of concepts but cleaned.
+ */
+function cleanAllConcept(allConcepts) {
+  var ret = [];
+  for (var concept in allConcepts) {
+    if (allConcepts[concept].python && allConcepts[concept].python != []) {
+      ret.push(allConcepts[concept]);
+    }
+  }
+  return ret;
+}
+
+/**
+ * This function allow us to find a concept by his function name.
+ * The function name is in the python list of a concept.
+ * @param cleanedConcepts The list of all the concepts, previously cleaned
+ * with cleanAllConcept(AllConcept)
+ * @param functionName The name of the function we have to look for
+ * @return A concept if found, false otherwise.
+ */
+function findConceptByFunction(cleanedConcepts, functionName) {
+  for (var concept in cleanedConcepts) {
```

Comment:
The variable names `concept` and `conceptFunctionName` are misleading : the variable concept will not contain a concept itself, just the list index of that concept in cleanedConcepts. A variable name such as "conceptIdx" would be more accurate.

Reply:
Yes you are right, thank you.

I thought it was the same behavior as java or other language when you do for (var ... in ...), but it only gets an id. I will think about that next time